tests/financial/test_national_stock_exchange_of_india.py

```diff
@@ -0,0 +1,742 @@
+
+    def test_republic_day(self):
+        name = "Republic Day"
+        self.assertNonObservedHolidayName(name, (f"{year}-01-26" for year in range(2001, 2050)))
+        no_dt = ("2025-01-26",)
+        self.assertNoHolidayName(name, no_dt)
+
+    def test_good_friday(self):
+        name = "Good Friday"
+        dt = (
+            "2022-04-15",
+            "2023-04-07",
+            "2024-03-29",
+            "2025-04-18",
+        )
+        self.assertHolidayName(name, dt)
+
+    def test_dr_baba_saheb_ambedkar_jayanti(self):
+        name = "Dr. Baba Saheb Ambedkar Jayanti"
+        dt = (
+            "2022-04-14",
+            "2023-04-14",
+            "2025-04-14",
+            "2026-04-14",
+        )
+        self.assertHolidayName(name, dt)
+        no_dt = ("2024-04-14",)
+        self.assertNoHoliday(no_dt)
+
+    def test_maharashtra_day(self):
+        name = "Maharashtra Day"
+        self.assertNonObservedHolidayName(name, (f"{year}-05-01" for year in range(2001, 2050)))
+        dt = ("2022-05-01",)
+        self.assertNoHoliday(dt)
+
+    def test_independence_day(self):
+        name = "Independence Day"
+        self.assertNonObservedHolidayName(name, (f"{year}-08-15" for year in range(2001, 2050)))
+
+    def test_mahatma_gandhi_jayanti(self):
+        name = "Mahatma Gandhi Jayanti"
+        self.assertNonObservedHolidayName(name, (f"{year}-10-02" for year in range(2001, 2050)))
+        dt = ("2022-10-02",)
+        self.assertNoHoliday(dt)
+
+    def test_christmas_day(self):
+        name = "Christmas Day"
+        self.assertNonObservedHolidayName(name, (f"{year}-12-25" for year in range(2001, 2050)))
+        dt = ("2022-12-25",)
+        self.assertNoHoliday(dt)
+
+    def test_mahashivratri(self):
+        name = "Maha Shivaratri"
+        dt = (
+            "2001-02-21",
+            "2002-03-12",
+            "2003-03-01",
+            "2004-02-18",
+            "2005-03-08",
+            "2006-02-26",
+            "2007-02-16",
+            "2008-03-06",
+            "2009-02-23",
+            "2010-02-12",
+            "2011-03-02",
+            "2012-02-20",
+            "2013-03-10",
+            "2014-02-27",
+            "2015-02-17",
+            "2016-03-07",
+            "2017-02-24",
+            "2018-02-13",
+            "2019-03-04",
+            "2020-02-21",
+            "2021-03-11",
+            "2022-03-01",
+            "2023-02-18",
+            "2024-03-08",
+            "2025-02-26",
+            "2026-02-15",
+            "2027-03-06",
+            "2028-02-23",
+            "2029-02-11",
+            "2030-03-02",
+            "2031-02-20",
+            "2032-03-10",
+            "2033-02-27",
+            "2034-02-17",
+            "2035-03-08",
+        )
+        self.assertNonObservedHolidayName(name, dt)
+        self.assertNonObservedHolidayName(name, range(2001, 2036))
+        no_dt = ("2023-02-18",)
+        self.assertNoHolidayName(name, no_dt)
```

Comment:
```suggestion
        self.assertNonObservedHolidayName(
            name,
            "2022-03-01",
            "2023-02-18",
            "2024-03-08",
            "2025-02-26",
        )
        self.assertNonObservedHolidayName(name, range(2001, 2036))
        self.assertNoHoliday(
            "2023-02-18",
        )
```
And the same for rest Hindu calendar holidays.

Reply:
updated.